ngraph/core/include/ngraph/op/constant.hpp

```diff
@@ -335,8 +363,52 @@ namespace ngraph
                     m_alloc_buffer_on_visit_attributes = val;
                 }
 
-            protected:
-                template <element::Type_t Type, typename OUT_T>
+            private:
+                template <element::Type_t Type,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type != element::Type_t::u1 &&
+                                                      Type != element::Type_t::u4 &&
+                                                      Type != element::Type_t::i4,
+                                                  bool>::type = true>
+                DataStorageType get_element_value(size_t index) const
+                {
+                    return get_data_ptr<Type>()[index];
```

Comment:
This will probably not segfault but there's a risk of an out-of-bounds read which might return some garbage if the index value is big enough. Can this be safely done like this, without any checks?

Reply:
@tomdol , @ilyachur,
Is doable but each time we need `shape_size(m_shape)` to check index range. Probably this value should be cache somewhere but solution required another analysis how to provide this.

What scary me more is espousing internal data by raw pointer: 
https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/include/ngraph/op/constant.hpp#L303-L313 and 
[guessing](https://github.com/openvinotoolkit/openvino/pull/5311/files#diff-0b193f8712af0a8bb0af386940481aac6d8b3f1bad91d529dbb69245f2d9658bR28) how big internal buffer is.